contracts/impl/KODAV2Override.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: knownorigin.io
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "../specs/IKODAV2Override.sol";
+
+/**
+ * @dev Implementation of KODA V2 override
+ * @notice KnownOrigin V2 (KODAV2) records expected commissions in simplistic single digit commission i.e. 100 = 10.0%, we dont store
+ *         a primary vs secondary expected commission amount so we need work this out proportionally
+ */
+contract KODAV2Override is IKODAV2Override, ERC165, Ownable {
+
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165) returns (bool) {
+        return interfaceId == type(IKODAV2Override).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /// @notice precision 100.00000%
+    uint256 public modulo = 100_00000;
+
+    /// @notice Secondary sale royalty fee
+    uint256 public creatorRoyaltiesFee = 10_00000; // 10% by default
+
+    function getKODAV2RoyaltyInfo(address _tokenAddress, uint256 _id, uint256 _amount)
+    external
+    override
+    view
+    returns (address payable[] memory receivers, uint256[] memory amounts) {
+
+        // Get the edition the token is part of
+        uint256 _editionNumber = IKODAV2(_tokenAddress).editionOfTokenId(_id);
```

Comment:
I think this contract should store the real v2 address and then if `_tokenAddress` does not match, revert as you have no guarantee that you're being passed the correct address. it can be a hardcoded constant so don't even need a constructor for that

Reply:
The registry handles the mapping and we can set the NFT token to point to this so I dont think it needs to be hard coded, at least thats what I understood from it anyway